examples/rust/components/http-client/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![allow(clippy::missing_safety_doc)]
+wit_bindgen::generate!();
+
+use std::io::Read;
+
+use exports::wasi::http::incoming_handler::Guest;
+use wasi::http::types::*;
+
+#[derive(serde::Deserialize)]
+struct DogResponse {
+    message: String,
+}
+
+struct HttpServer;
+
+impl Guest for HttpServer {
+    fn handle(_request: IncomingRequest, response_out: ResponseOutparam) {
+        let req = wasi::http::outgoing_handler::OutgoingRequest::new(Fields::new());
+        req.set_scheme(Some(&Scheme::Https)).unwrap();
+        req.set_authority(Some("dog.ceo")).unwrap();
+        req.set_path_with_query(Some("/api/breeds/image/random"))
+            .unwrap();
+        let dog_picture_url = match wasi::http::outgoing_handler::handle(req, None) {
+            Ok(resp) if wasi::io::poll::poll(&[&resp.subscribe()]) == [0] => {
+                let response = resp
+                    .get()
+                    .expect("HTTP request response missing")
+                    .expect("HTTP request response requested more than once")
+                    .expect("HTTP request failed");
+                if response.status() == 200 {
+                    let response_body = response
+                        .consume()
+                        .expect("failed to get incoming request body");
+                    let body = {
+                        let mut buf = vec![];
+                        let mut stream = response_body
+                            .stream()
+                            .expect("failed to get HTTP request response stream");
+                        InputStreamReader::from(&mut stream)
+                            .read_to_end(&mut buf)
+                            .expect("failed to read value from HTTP request response stream");
+                        buf
+                    };
+                    let _trailers = wasi::http::types::IncomingBody::finish(response_body);
+                    let dog_response: DogResponse = serde_json::from_slice(&body).unwrap();
+                    dog_response.message
+                } else {
+                    panic!("HTTP request failed with status code {}", response.status());
+                }
+            }
+            Ok(_) => {
+                panic!("Got response, but it wasn't ready");
```

Comment:
I don't think that's the right behavior for an example - we should be showing best practices here, why not block on the pollable until it's ready instead?

Reply:
I have no idea how to do that and this is copied from what the plugin does (it just logs an error) which is copied from what our test does. Is blocking just looping until it is ready or is there some other mechanism I can use?

Reply:
Figured it out. `Pollable` just has `block`. Updating in all examples